### PR TITLE
chore: :art: update eslint rule, remove prettier

### PR DESCRIPTION
**Pull request by 2214962083**
1. Add more and more perfect eslint rules to make the code more standardized and stronger

2. Remove prettier, use `eslint . --fix` instead of format code function

3. Put the ide configuration folder in `.gitignore`, allowing developers to configure their own ide without submitting it to the warehouse, and keep a copy of vscode example settings

### CODE REVIEW

File: src/hooks/useGenerateResult.ts [hunk 1]
```diff
@@ -16,15 +16,13 @@ export const useGenerateResult = () => {
       body: JSON.stringify(body),
     })
 
-    if (!response.ok) {
+    if (!response.ok)
```
Comment (futantan) [inline src/hooks/useGenerateResult.ts:19]: hi @2214962083 
Could you help to update the eslint rules to let the brackets after if statement a required one?

**Comment by futantan** [comment]
awesome work! @2214962083 
all looks good to me except the comment above. 
I accept all the code styling but the one may lead to the mistake. 

Thanks for your contribution!

**Review by futantan — CHANGES_REQUESTED** [review]

File: src/hooks/useGenerateResult.ts (hunk 1 above)
Comment (2214962083) [reply src/hooks/useGenerateResult.ts]: sure, i have push the latest code to fix it

**Review by futantan — APPROVED** [review]

**Review by futantan — APPROVED** [review]

**Comment by futantan** [comment]
hi @2214962083 
I found that after removing prettier, the code editor won't help me to format the code, for example the jsx content
Could you help to create another PR to add prettier back and compatible with eslint.

File: .vscode-example/settings.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+    "prettier.enable": false,
+    "editor.formatOnSave": true,
+    "editor.codeActionsOnSave": {
+      "source.fixAll.eslint": true
+    }
+}
```
Comment (2214962083) [inline .vscode-example/settings.json:7]: @futantan if your vscode don't format on save, you can copy vscode-example/settings.json to try, it's work for me

Comment (futantan) [reply .vscode-example/settings.json:7]: https://user-images.githubusercontent.com/6268441/224469317-851a6736-f94c-4063-958f-048d2830b408.mp4

any other steps needed?

Comment (2214962083) [reply .vscode-example/settings.json:7]: i try as your code, the format is working, do you install the eslint vscode extension or you can paste your vscode settings.json here

Comment (2214962083) [reply .vscode-example/settings.json:7]: or try restart the vscode

Comment (2214962083) [reply .vscode-example/settings.json:7]: dont forget run pnpm i

Comment (futantan) [reply .vscode-example/settings.json:7]: all tried, not working...

Uploading CleanShot 2023-03-11 at 14.59.37.mp4…

Comment (2214962083) [reply .vscode-example/settings.json:7]: ok i got it, i will  add prettier back

Comment (futantan) [reply .vscode-example/settings.json:7]: thank you! and I will revert the PR first  so that won't block my work and merge another one

**Pull request merged**